AK/StdLibExtraDetails.h

```diff
@@ -573,6 +573,8 @@ struct __decay<T[N]> {
 template<typename T>
 using Decay = typename __decay<T>::type;
 
+template<typename T, typename U>
+inline constexpr bool IsAnyPointerOf = IsPointer<Decay<U>>&& IsSame<T, RemoveCV<RemovePointer<Decay<U>>>>;
```

Comment:
A bit bikeshed-y, but `IsAnyPointerOf` sounds kind of awkward. Maybe `IsPointerOfType` or something?